### PR TITLE
Support bitronix and narayana transaction testing

### PR DESCRIPTION
**Pull request by FlyingZC**
Changes proposed in this pull request:
- Support bitronix and narayana transaction testing in both proxy and JDBC adatper

### CODE REVIEW

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/assembly/conf/jbossts-properties.xml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    <entry key="RecoveryEnvironmentBean.expiryScannerClassNames">
+        com.arjuna.ats.internal.arjuna.recovery.ExpiredTransactionStatusManagerScanner
+    </entry>
+    <entry key="RecoveryEnvironmentBean.recoveryPort">4712</entry>
+    <entry key="RecoveryEnvironmentBean.recoveryAddress"></entry>
+    <entry key="RecoveryEnvironmentBean.transactionStatusManagerPort">0</entry>
+    <entry key="RecoveryEnvironmentBean.transactionStatusManagerAddress"></entry>
+    <entry key="RecoveryEnvironmentBean.recoveryListener">NO</entry>
+    <entry key="RecoveryEnvironmentBean.recoveryBackoffPeriod">1</entry>
+</properties>
```
Comment (jingshanglu) [inline shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/assembly/conf/jbossts-properties.xml:42]: There should be a empty line at the end of file.

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-transaction/src/test/resources/jbossts-properties.xml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    <entry key="RecoveryEnvironmentBean.expiryScannerClassNames">
+        com.arjuna.ats.internal.arjuna.recovery.ExpiredTransactionStatusManagerScanner
+    </entry>
+    <entry key="RecoveryEnvironmentBean.recoveryPort">4712</entry>
+    <entry key="RecoveryEnvironmentBean.recoveryAddress"></entry>
+    <entry key="RecoveryEnvironmentBean.transactionStatusManagerPort">0</entry>
+    <entry key="RecoveryEnvironmentBean.transactionStatusManagerAddress"></entry>
+    <entry key="RecoveryEnvironmentBean.recoveryListener">NO</entry>
+    <entry key="RecoveryEnvironmentBean.recoveryBackoffPeriod">1</entry>
+</properties>
```
Comment (jingshanglu) [inline shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-transaction/src/test/resources/jbossts-properties.xml:42]: There should be a empty line at the end of file.

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/assembly/conf/jbossts-properties.xml (hunk 1 above)
Comment (FlyingZC) [reply shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/assembly/conf/jbossts-properties.xml:42]: Your are right.

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-transaction/src/test/resources/jbossts-properties.xml (hunk 2 above)
Comment (FlyingZC) [reply shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-transaction/src/test/resources/jbossts-properties.xml:42]: Your are right.

File: .github/workflows/it-transaction.yml [hunk 3]
```diff
@@ -20,18 +20,14 @@ name: Transaction Integration Test
...
   push:
     branches: [ master, dev ]
-    paths:
-      - '!*.md'
+
   pull_request:
     branches: [ master ]
     paths:
       - '!examples/**'
-      - '.github/workflows/it-transaction.yml'
```
Comment (jingshanglu) [inline .github/workflows/it-transaction.yml:29]: Hi, why delete this line?

Comment (FlyingZC) [reply .github/workflows/it-transaction.yml:29]: I merged from master.

**Review by jingshanglu — APPROVED** [review]

**Pull request merged**